src/components/dev-hub/podcast-player/player-reducer.js

```diff
@@ -0,0 +1,29 @@
+export const PLAYER_ACTIONS = {
+    SET_DURATION: 'setDuration',
+    SET_IS_READY: 'setIsReady',
+    SET_PROGRESS: 'setProgress',
+    TOGGLE_IS_PLAYING: 'toggleIsPlating',
+};
+
+export const playerInitialState = {
+    duration: null,
+    isPlaying: false,
+    isReady: false,
+    progress: { playedSeconds: 0 },
+};
+
+export const playerReducer = (state, { type, value }) => {
```

Comment:
This is awesome, very well done! We can incorporate this logic into the existing player someday :)